utils_cv/detection/dataset.py

```diff
@@ -273,11 +280,55 @@ def split_train_test(
         train = copy.deepcopy(Subset(self, indices[test_num:]))
         train.dataset.transforms = self.train_transforms
 
-        test = copy.deepcopy(Subset(self, indices[: test_num + 1]))
+        test = copy.deepcopy(Subset(self, indices[: test_num]))
         test.dataset.transforms = self.test_transforms
 
         return train, test
 
+    def init_data_loaders(self):
+        """ Create training and validation data loaders """
+        self.train_dl = DataLoader(
+            self.train_ds,
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=db_num_workers(),
+            collate_fn=collate_fn,
+        )
+
+        self.test_dl = DataLoader(
+            self.test_ds,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=db_num_workers(),
+            collate_fn=collate_fn,
+        )
+
+    def add_images(self, im_paths: List[str], anno_bboxes: List[AnnotationBbox], target: str = "train"):
```

Comment:
why does this function take in anno_bboxes? in the notebook, wont all the gt anno_bboxes for these just be the tiny rectangle in the top let?

maybe this function should be called `add_negative_images` or something less generic.

Reply:
This function can be used to add more negative or positive images. Hence the anno_bboxes as required input.